powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/AppConfigProvider.java

```diff
@@ -0,0 +1,212 @@
+import software.amazon.awssdk.services.appconfigdata.model.GetLatestConfigurationRequest;
+import software.amazon.awssdk.services.appconfigdata.model.GetLatestConfigurationResponse;
+import software.amazon.awssdk.services.appconfigdata.model.StartConfigurationSessionRequest;
+import software.amazon.lambda.powertools.parameters.cache.CacheManager;
+import software.amazon.lambda.powertools.parameters.transform.TransformationManager;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implements a {@link ParamProvider} on top of the AppConfig service. AppConfig provides
+ * a mechanism to retrieve and update configuration of applications over time.
+ * AppConfig requires the user to create an application, environment, and configuration profile.
+ * The configuration profile's value can then retrieved, by key name, through this provider.
+ *
+ * Because AppConfig is designed to handle rollouts of configuration over time, we must first
+ * establish a session for each key we wish to retrieve, and then poll the session for the latest
+ * value when the user re-requests it. This means we must hold a keyed set of session tokens
+ * and values.
+ *
+ * @see <a href="https://awslabs.github.io/aws-lambda-powertools-java/utilities/parameters">Parameters provider documentation</a>
+ * @see <a href="https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-working.html">AppConfig documentation</a>
+ */
+public class AppConfigProvider extends BaseProvider{
+
+    private static class EstablishedSession {
+        private final String nextSessionToken;
+        private final String lastConfigurationValue;
+
+        private EstablishedSession(String nextSessionToken, String value) {
+            this.nextSessionToken = nextSessionToken;
+            this.lastConfigurationValue = value;
+        }
+    }
+
+    private final AppConfigDataClient client;
+
+    private final String application;
+
+    private final String environment;
+
+    private final HashMap<String, EstablishedSession> establishedSessions = new HashMap<>();
+
+    public AppConfigProvider(CacheManager cacheManager, String environment, String application) {
+        this(cacheManager, AppConfigDataClient.builder()
+                        .httpClientBuilder(UrlConnectionHttpClient.builder())
+                        .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+                        .region(Region.of(System.getenv(SdkSystemSetting.AWS_REGION.environmentVariable())))
+                        .build(),
+                environment, application);
+    }
+
+    AppConfigProvider(CacheManager cacheManager, AppConfigDataClient client, String environment, String application) {
+        super(cacheManager);
+        this.client = client;
+        this.application = application;
+        this.environment = environment;
+    }
+
+
+    @Override
+    protected String getValue(String key) {
```

Comment:
Hello @scottgerring! I might suggest renaming the name of this parameter to `profile` instead of `key`. 

AppConfig uses the idea of `profile`  which can be a JSON/TXT/YAML or even a different source like S3.

Reply:
Hey @leandrodamascena I went to change this and then realized it is the name of the field in the base class - `    protected abstract String getValue(String key);`. Because it's used in terms of the common interface I've added javadoc to the method to explain that it maps back to an AppConfig `profile` in this case